lib/widgets/todo_items.dart

```diff
@@ -46,6 +51,27 @@ class _TodoItemsState extends State<TodoItems> {
         controlAffinity: ListTileControlAffinity.leading,
         onChanged: (bool completedTask) => setState(() => _isCompleted = completedTask),
         tileColor: _isCompleted ? isCompletedTheme : isNotCompletedTheme,
+        secondary: IconButton(
+            icon: Icon(
+              Icons.edit,
+              color: primaryTheme,
+            ),
+            onPressed: () {
+               return showModalBottomSheet(
+              isScrollControlled: true,
+              context: context,
+              builder: (BuildContext context) {
+                return GestureDetector(
+                  onTap: () {},
+                  child: AddTodo(
+                    (String newTodo, String newDate) => provider.addNewTodo(newTodo, newDate),
```

Comment:
What is this for when you are already calling the `addNewTodo` from AddTodo Screen?

Reply:
I will remove functions that are not needed in my PR Feedback.

Reply:
Think about it. When you are going to edit the existing todo, everything happens inside the AddTodo screen (add and update of todo). So, it is quite redundant doing it here since you are doing it there already.

Reply:
Besides, the TodoProvider from Overview screen is already listening to changes.